### PR TITLE
Add published version of Planck 18 cosmology.

### LINKED ISSUE #9515: MNT: Rename Planck 2018 when paper is accepted
### Description

A direct follow-up of #8111. 

I am tentatively milestoning this to 4.1 so we remember to revisit this in the next release after 4.0. If necessary, please adjust the milestone.

### Additional context

Also see #8006

### PR DESCRIPTION
**Pull request by aconley**
### Description

The day has finally arrived, the paper is accepted and has a proper citation.

Fixes #9515

### CODE REVIEW

**Comment by aconley** [comment]
There was a part of me that was tempted to make it the default cosmology, but we should probably let it be in the wild for a little bit first.

File: CHANGES.rst
Comment (pllim) [inline CHANGES.rst:1598]: Instead of modifying change log of an older release, this should be a new change log entry under 4.2.

File: astropy/cosmology/parameters.py
Comment (pllim) [inline astropy/cosmology/parameters.py:78]: Do we have to actually emit a deprecation warning if this is used?

File: docs/cosmology/index.rst
Comment (pllim) [inline docs/cosmology/index.rst:305]: The table has to align. This produced a warning in RTD. In this case, the column with `67.66` also has to be expanded from 4-char to 5-char.

**Review by pllim — COMMENTED** [review]
Thank you for the quick turnaround!

I am just leaving a general review. Someone who is more familiar with cosmology should review the content.

**Comment by pllim** [comment]
Question for maintainers: Do we need What's New for this? If not, feel free to remove the relevant label.

File: astropy/cosmology/parameters.py
Comment (aconley) [reply astropy/cosmology/parameters.py:78]: I don't know.  It seems like it would be nice -- but how?  The existing deprecation annotations are for classes and attributes -- but this is neither.

Comment (pllim) [reply astropy/cosmology/parameters.py:78]: If it is passed into `default_cosmology`, then perhaps a warning could be emitted there, but I am not familiar with this sub-package.

File: astropy/cosmology/parameters.py
Comment (pllim) [inline astropy/cosmology/parameters.py:194]: I also wonder if we should remove the deprecated one from here.
```suggestion
available = ['Planck18', 'Planck15', 'Planck13', 'WMAP9', 'WMAP7',
```

**Comment by bsipocz** [comment]
Is there any downside of making this a default now and not in a future release?

**Comment by pllim** [comment]
> Is there any downside of making this a default now and not in a future release?

Other than changing existing default answers... 🤷

File: astropy/cosmology/parameters.py
Comment (aconley) [reply astropy/cosmology/parameters.py:194]: That would just remove it.  The cosmology objects are built off of that list, so import Planck18_arXiv_v2 would stop working.

File: astropy/cosmology/parameters.py
Comment (aconley) [reply astropy/cosmology/parameters.py:78]: Me neither.

Who is the expert on astropy deprecation?

**Comment by aconley** [comment]
The downside is that we haven't warned people, so it's pure conservatism not to.  But if you both think it's not a big deal, it probably is what most people would have guessed was the default anyways.

Of course, anybody that actually cares about that level of detail really shouldn't be using the default cosmology anyways!

File: astropy/cosmology/parameters.py
Comment (pllim) [reply astropy/cosmology/parameters.py:194]: Oh, okay. Sorry for the noise!

**Comment by pllim** [comment]
Or we can keep it as-is (non-default) and make a note that we will make it default in 4.3? And then we make it default as soon as 4.2 is branched to allow plenty of time for people to test against dev.

As for deprecation, I don't think there is a precedent in deprecating cosmology. It is just that mentioning it in comment without any implementation isn't visible to users.

**Comment by pllim** [comment]
@aconley , in order to push things forward quickly, there are the only 2 real blockers. The rest we can discuss post-merge and still it would be okay.

1. Fix change log.
2. Fix table formatting so RTD job passes.

File: CHANGES.rst
Comment (aconley) [reply CHANGES.rst:1598]: Oh, right, sorry, missed the section heading.

File: astropy/cosmology/parameters.py
Comment (aconley) [reply astropy/cosmology/parameters.py:78]: Added DEPRECATED to the docstring.

There's no equals method for cosmology objects implemented.  Could check the docstring, but that feels a bit wierd.

**Comment by aconley** [comment]
Done with the minimal changes, will try to grab some time to change the default later tonight.

File: docs/cosmology/index.rst
Comment (aconley) [reply docs/cosmology/index.rst]: Fixed.

**Comment by aconley** [comment]
Made it the default.  Surprisingly painless.

**Comment by aconley** [comment]
Test failures seem unrelated -- some sort of cache clearing problem.

**Review by pllim — COMMENTED** [review]
Thanks! I think this is very close now. Just minor comments on how to emit deprecation in code and an extra line to change log for consistency.

File: CHANGES.rst
Comment (aconley) [reply CHANGES.rst]: Done, slightly different wording.

File: astropy/cosmology/core.py
Comment (aconley) [reply astropy/cosmology/core.py]: Done, omitted cosmology because it's redundant in context (message says [astropy.cosmology.core]).

**Review by pllim — APPROVED** [review]
I took liberty to push a couple of small fixes directly to this PR, I hope you don't mind. LGTM now and thanks for including a what's new entry!

**Review by bsipocz — APPROVED** [review]

**Comment by bsipocz** [comment]
Thank you both guarding this through this quickly!

**Pull request merged**
